orc8r/cloud/go/orc8r/math/math.go

```diff
@@ -15,16 +15,34 @@ package math
 
 import "hash/fnv"
 
-// JitterUint32 returns a jitter of the given uint32 value that is deterministic
-// based on the given key.
-func JitterUint32(n uint32, key string, maxMultiplier float32) uint32 {
+func getFNV1Hash(key string) uint32 {
 	// FNV-1 is a non-cryptographic hash function that is fast and very simple to implement
 	h := fnv.New32a()
 	_, err := h.Write([]byte(key))
 	if err != nil {
 		return 0
 	}
-	multiplier := float32(h.Sum32()%100) / 100.0
-	maxJitter := float32(n) * maxMultiplier
-	return uint32(multiplier * maxJitter)
+	return h.Sum32()
+}
+
+// JitterUint32 returns a jitter of the given uint32 value that is deterministic
+// based on the given key.
+func JitterUint32(n uint32, key string, maxMultiplier float32) uint32 {
+	fnv1Hash := getFNV1Hash(key)
+	multiplier := maxMultiplier * float32(fnv1Hash%100) / 100.0
+	if multiplier >= 1 {
+		return n
+	}
+	return uint32(multiplier * float32(n))
```

Comment:
Here and below: this isn't quite what we want, actually should be something like

```suggestion
	multiplier := maxMultiplier * float32(hash%100) / 100.0
	jittered := n + (n * multiplier)

	// Check for integer overflow
	if jittered < n {
		return n
	}
	
	return jittered
```

Reply:
I thought the function is supposed to return a jitter (a fraction of `n`) instead of `n + jitter`, according to its original implementation? @hcgatewood

Reply:
AH yes, I see. So you're correct, it's currently implemented+used as `return jitter`. That said, I'd originally envisioned it would be `return n + jitter`). And I think the latter is preferable because, like I mention, we can handle the case where there's integer overflow

Up to you though, if you wanna leave it that's fine with me